modules/web/js/ballerina/components/transform-statement-decorator.jsx

```diff
@@ -350,99 +380,97 @@ class TransformStatementDecorator extends React.Component {
-                        conRight.targetProperty = complexTargetProp.names.reverse();
-                        conRight.isComplexMapping = false;
-                        self.mapper.addConnection(conRight);
+                if (func.getReturnParams().length !== leftExpressions.getChildren().length) {
+                    alerts.warn('Function inputs and mapping count does not match in "' + func.getName() + '"');
+                } else {
+                    let funcSource = this.getConnectionProperties('source', rightExpression);
+                    _.forEach(leftExpressions.getChildren(), (expression, i) => {
+                        let source = this.getConnectionProperties('source', func.getReturnParams()[i]);
+                        _.merge(source, funcSource); //merge parameter props with function props
+                        let target = this.getConnectionProperties('target', expression);
+                        this.drawConnection(rightExpression.getID(), source, target);
                     });
-                } else { 
-                     alerts.warn('Function outputs and mapping count does not match in "' + func.getName() + '"');
                 }
+            } else {
+                log.error('Invalid expression type in transform statement body');
+            }
+        } else {
+            log.error('Invalid statement type in transform statement');
+        }
+    }
+
+    getConnectionProperties(type, expression) {
+        let con = {};
+        if (BallerinaASTFactory.isFieldAccessExpression(expression)) {
+            con[type + 'Struct'] = expression.getChildren()[0].getVariableName();
+            let complexProp = this.createComplexProp(con[type + 'Struct'],
+                                        expression.getChildren()[1].getChildren());
+            con[type + 'Type'] = complexProp.types.reverse();
+            con[type + 'Property'] = complexProp.names.reverse();
+        } else if (BallerinaASTFactory.isFunctionInvocationExpression(expression)) {
+            con[type + 'Function'] = true;
+            con[type + 'Struct'] = expression.getPackageName() + '-' + expression.getFunctionName();
+            con[type + 'Id'] = expression.getID();
+        } else if (BallerinaASTFactory.isVariableReferenceExpression(expression)) {
+            con[type + 'Struct'] = expression.getVariableName();
+            let varRef = _.find(self.predefinedStructs, { name:expression.getVariableName()});
+            if (!_.isUndefined(varRef)) {
+                con[type + 'Type'] = [varRef.type];
             }
+            con[type + 'Property'] = [expression.getVariableName()];
+        } else if (['name','type'].every(prop => prop in expression)) {
+            con[type + 'Property'] = [expression.name];
+            con[type + 'Type'] = [expression.type];
+        } else if (_.has(expression, 'type')) {
+            con[type + 'Property'] = [undefined];
+            con[type + 'Type'] = [expression.type];
```

Comment:
We will move all these string constants to a different module with a later commit (since there is a lot ATM)